### PR TITLE
Add DISTINCT ON capability via :distinct_on option

### LINKED ISSUE #3: Using SELECT DISTINCT ON attribute
using psql's DISTINCT ON requires the columns to be selected in the same order as the distinct

Since order_as_specified insert a new ORDER BY, I can't use  SELECT DISTINCT ON...

The only way I see is to use the same method for generating the order by, outside of the gem. 

Is there a way to use this gem with queries that involve DISTINCT ON?

### PR DESCRIPTION
**Pull request by JacobEvelyn**
Closes #3.

### CODE REVIEW

File: spec/config/test_setup_migration.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    create_table :test_classes do |t|
+      t.string :field
+    end
+
+    create_table :association_test_classes do |t|
+      t.integer :test_class_id
+    end
+  end
+
+  def down
```
Comment (davidlago) [inline spec/config/test_setup_migration.rb]: Where do we use this method?

Comment (JacobEvelyn) [inline spec/config/test_setup_migration.rb]: We don't; it's just nice to be able to undo the migration. I modeled this code heavily off of what we do in [unique_attributes](https://github.com/panorama-ed/unique_attributes/tree/master/spec).

**Comment by davidlago** [comment]
👍 LGTM. I don't recall what the process for approving this PR was... should I mark it as approved in some other way?

**Comment by JacobEvelyn** [comment]
Historically we'd give some sort of "LGTM" comment and then assign back to the author. Thanks for the review!

**Pull request merged**